packages/data-point/lib/debug-utils/stack.test.js

```diff
@@ -0,0 +1,439 @@
+    })
+}
+
+const throwIfEquals = v1 => v2 => {
+  return v1 === v2 ? throwError() : false
+}
+
+nock('http://remote.test')
+  .get('/')
+  .reply(200, {
+    ok: true
+  })
+
+nock('http://remote.test')
+  .get('/source1')
+  .reply(404, 'not found')
+
+const dataPoint = DataPoint.create({
+  entities: {
+    'transform:1': () => {
+      throw new Error('test error')
+    },
+    'transform:2': 'transform:1',
+    'transform:3': 'transform:2',
+
+    'request:1': {
+      value: [identity, throwError],
+      url: 'http://remote.test'
+    },
+    'request:2': {
+      options: {
+        x: () => 'apples',
+        y: [identity, throwError]
+      },
+      url: 'http://remote.test'
+    },
+    'request:3': {
+      before: throwError,
+      url: 'http://remote.test'
+    },
+    'request:4': {
+      url: 'http://remote.test',
+      after: throwError
+    },
+    'request:5': {
+      url: 'http://remote.test/source1'
+    },
+
+    // fail first case
+    'control:1': {
+      select: [
+        { case: throwError, do: throwError },
+        { case: throwError, do: throwError },
+        { default: throwError }
+      ]
+    },
+    // fail second case
+    'control:2': {
+      select: [
+        { case: _false, do: throwError },
+        { case: throwError, do: throwError },
+        { default: throwError }
+      ]
+    },
+    // fail first do
+    'control:3': {
+      select: [
+        { case: _true, do: throwError },
+        { case: throwError, do: throwError },
+        { default: throwError }
+      ]
+    },
+    // fail second do
+    'control:4': {
+      select: [
+        { case: _false, do: throwError },
+        { case: _true, do: throwError },
+        { default: _true }
+      ]
+    },
+    // fail default
+    'control:5': {
+      select: [
+        { case: _false, do: throwError },
+        { case: _false, do: throwError },
+        { default: throwError }
+      ]
+    },
+
+    'model:1': {
+      before: throwError
+    },
+    'model:2': {
+      value: throwError
+    },
+    'model:3': {
+      after: throwError
+    },
+    'model:4': {
+      value: throwError,
+      error: throwError
+    },
+
+    'entry:1': {
+      before: throwError
+    },
+    'entry:2': {
+      value: throwError
+    },
+    'entry:3': {
+      after: throwError
+    },
+    'entry:4': {
+      value: throwError,
+      error: throwError
+    },
+
+    'entry:type-check-1': {
+      inputType: 'string'
+    },
+    'entry:type-check-2': {
+      value: () => 500,
+      outputType: 'string'
+    },
+    'entry:type-check-3': {
+      inputType: 'schema:a.1.0'
+    },
+    'entry:type-check-4': {
+      value: () => 500,
+      outputType: 'schema:a.1.0'
+    },
+
+    'schema:with-value-prop': {
+      value: throwError,
+      schema: {
+        type: 'object'
+      }
+    }
+  }
+})
+
+dataPoint.addEntities(schemaA10)
+
+describe('transform entity stack traces', () => {
+  test('transform:1', () => {
```

Comment:
The descriptions could use a little improvement (even though I expect they'll all say something similar). Just thinking that if this is run with `--verbose` there's no additional information from the descriptions.

Reply:
I went through and added better descriptions; please take a look when you have a chance